src/components/SwapForm/service/getBalanceNear.ts

```diff
@@ -0,0 +1,35 @@
+import { formatUnits } from "viem"
+
+import { nearAccount } from "@src/utils/near"
+import {
+  NEAR_MIN_RESERVE_BALANCE,
+  NEAR_STORAGE_COST_PRE_BYTE,
+} from "@src/constants/contracts"
+
+export const minimumNearBalance = (storageUsed: number): number => {
+  const storageCostPerByte = NEAR_STORAGE_COST_PRE_BYTE
+  const calculateMinBalance = storageUsed * storageCostPerByte
+  return storageUsed <= 770
+    ? 0
+    : calculateMinBalance > NEAR_MIN_RESERVE_BALANCE
+      ? calculateMinBalance
+      : NEAR_MIN_RESERVE_BALANCE
+}
+
+export const getBalanceNear = async (accountId: string) => {
+  try {
+    const viewAccount = await nearAccount(accountId)
+    if (!viewAccount) {
+      return 0
+    }
+    const formattedAmountOut = formatUnits(BigInt(viewAccount.amount), 24)
+
+    return (
```

Comment:
The operations with numbers should be done via BigInt otherwise we are losing precision.
We can convert to more human friendly format in the UI level

Reply:
This function works close to the user interface level and therefore it is more useful to do these transformations here in place, since it compares the user input and the input is not used precision. That's why here we are not losing anything.

Reply:
You ll end up in a situation where you may want to reuse it but It is going to be already converted to number and the precision will be lost.

I used to do the same thing and been corrected by other ppl.

The main point - the service function that get onchain data should operate with big number, all of the formatting should be made outside even it is an extra step.

P.S. You can leave it as it is if you want to

Reply:
This isn't relevant case cause we will not use this function in other places, we need this function only for calculation what is the maximun near is available to swap and as you can see result it isn't pure user near balance, here `return (
      parseFloat(formattedAmountOut) -
      minimumNearBalance(viewAccount.storage_usage)
    )`, so this isn't the case.

Reply:
I will rename it to be more precisely what this function does.